src/core/net/udp6.cpp

```diff
@@ -527,6 +527,18 @@ void Udp::HandlePayload(Message &aMessage, MessageInfo &aMessageInfo)
     return;
 }
 
+bool Udp::IsPortInUse(uint16_t aPort) const
+{
+    for (const SocketHandle * socket = mSockets.GetHead(); socket != nullptr; socket=socket->GetNext())
```

Comment:
Currently there are 4 kinds of UDP sockets:
1. Pure Thread stack (non-Platform_UDP) sockets (e.g. MLE, TMF)
2. Sockets bound to All Netifs (i.e. OT_NETIF_UNSPECIFIED)
3. Sockets bound Thread Netif (i.e. OT_NETIF_THREAD)
4. Sockets bound to Bacbkone Netif (i.e. OT_NETIF_BACKBONE)

I think we actually needs to pass the UDP messages to Host for case 4 because in case 4 the UDP port is not actually used by OT. OT bound the UDP socket to the Backbone netif, however there might be an user APP that bound to the same port but on Thread netif only. That APP should be able to receive the UDP messages using POSIX socket APIs.

Reply:
Any idea how to solve all the possible cases?
I was wondering about instead of automatic filtering all ports in use allow application to pass list of ports to filter out - if app is using OpenThread API for communication it usually is aware of it ;)

Reply:
> Any idea how to solve all the possible cases?

As I have described, we need to skip case 4. BTW, we can use`Udp::IsBackboneSocket` to check if a socket is of case 4.

> I was wondering about instead of automatic filtering all ports in use allow application to pass list of ports to filter out - if app is using OpenThread API for communication it usually is aware of it ;)

I think automatic filtering should work.